addons/sale/models/account_move.py

```diff
@@ -28,6 +28,19 @@ def action_post(self):
                 pass
         return res
 
+    def button_draft(self):
+        #inherit of the function from account.move to reverse the computation made by action_post
+        super(AccountMove, self).button_draft()
+        line_ids = self.mapped('line_ids').filtered(lambda line: line.sale_line_ids.is_downpayment)
+        for line in line_ids:
+            try:
+                if not all(line.tax_ids.mapped('price_include')):
+                    line.sale_line_ids.price_unit = -line.sale_line_ids.untaxed_amount_to_invoice
+            except UserError:
+                # a UserError here means the SO was locked, which prevents changing the taxes
+                # just ignore the error - this is a nice to have feature and should not be blocking
+                pass
```

Comment:
You can't do that like this. Taxes are not so "easy". This whole downpayment feature is buggy regarding its taxes management so I think we should consider this as a limitation and let the user fixes the invoice manually. Someone in accounting is improving that.

Reply:
Ok, seems to be already fixed since I pushed anyway, thanks !